dashboard/lib/tasks/asset_sync.rake

```diff
@@ -0,0 +1,11 @@
+namespace :assets do
+  desc 'Synchronize assets to S3'
+  task :sync => :environment do
+    # AWS CLI implements an optimized `sync` utility without any Ruby SDK equivalent.
+    `aws s3 sync #{CDO.dashboard_dir}/public/assets s3://#{CDO.assets_bucket}/#{rack_env}/assets --acl public-read --cache-control 'max-age=31536000'`
+  end
+end
+
+Rake::Task['assets:precompile'].enhance do
+  Rake::Task['assets:sync'].invoke if CDO.daemon
```

Comment:
`assets:precompile` isn't normally used in development (`config.assets.digest` is default `false` and `config.assets.compile` is default `true` in `development`), and is skipped in our `rake build`. If `rake assets:precopile` is run manually, then yes it would attempt to sync to `/development` on s3, but the assets would still be served locally.